skimage/feature/peak.py

```diff
@@ -15,60 +15,111 @@ def _get_high_intensity_peaks(image, mask, num_peaks):
-    for i, excluded in enumerate(exclude_border):
-        if excluded == 0:
+    # zero out label borders
+    for i, width in enumerate(border_width):
+        if width == 0:
             continue
-        mask[(slice(None),) * i + (slice(None, excluded),)] = False
-        mask[(slice(None),) * i + (slice(-excluded, None),)] = False
-    return mask
+        label[(slice(None),) * i + (slice(None, width),)] = 0
+        label[(slice(None),) * i + (slice(-width, None),)] = 0
+    return label
+
+
+def _get_threshold(image, threshold_abs, threshold_rel):
+    """Return the threshold value according to an absolute and a relative
+    value.
+
+    """
+    threshold = threshold_abs if threshold_abs is not None else image.min()
+
+    if threshold_rel is not None:
+        threshold = max(threshold, threshold_rel * image.max())
+
+    return threshold
+
+
+def _get_excluded_border_width(image, min_distance, exclude_border):
+    """Return border_width values relative to a min_distance if requested.
+
+    """
+
+    if isinstance(exclude_border, bool):
+        border_width = (min_distance if exclude_border else 0,) * image.ndim
+    elif isinstance(exclude_border, int):
+        if exclude_border < 0:
+            raise ValueError("`exclude_border` cannot be a negative value")
+        border_width = (exclude_border,) * image.ndim
+    elif isinstance(exclude_border, tuple):
+        if len(exclude_border) != image.ndim:
+            raise ValueError(
+                "`exclude_border` should have the same length as the "
+                "dimensionality of the image.")
+        for exclude in exclude_border:
+            if not isinstance(exclude, int):
+                raise ValueError(
+                    "`exclude_border`, when expressed as a tuple, must only "
+                    "contain ints."
+                )
+            if exclude < 0:
+                raise ValueError(
+                    "`exclude_border` can not be a negative value")
+        border_width = exclude_border
+    else:
+        raise TypeError(
+            "`exclude_border` must be bool, int, or tuple with the same "
+            "length as the dimensionality of the image.")
+
+    return border_width
 
 
 @remove_arg("indices", changed_version="0.20")
 def peak_local_max(image, min_distance=1, threshold_abs=None,
                    threshold_rel=None, exclude_border=True, indices=True,
                    num_peaks=np.inf, footprint=None, labels=None,
-                   num_peaks_per_label=np.inf):
+                   num_peaks_per_label=np.inf, p_norm=np.inf):
     """Find peaks in an image as coordinate list or boolean mask.
 
     Peaks are the local maxima in a region of `2 * min_distance + 1`
     (i.e. peaks are separated by at least `min_distance`).
 
-    If there are multiple local maxima with identical pixel intensities
-    inside the region defined with `min_distance`,
-    the coordinates of all such pixels are returned.
-
```

Comment:
Why is this behaviour changed, @rfezzani?

Reply:
At any rate, the behaviour in plateaus should be explained in the docstring, whatever it is.

Reply:
Thank you for your review @jni. This behavior is exactly what I wanted to fix in this PR as it is reported in #2592. This explains all the reported changes in the tests below :wink:.

Reply:
@rfezzani I've pushed a note to this effect, please pull before any further changes!

Reply:
Cool, thx @jni :wink: